### ISSUE TITLE
Cannot read the `content` attribute of the `google_storage_bucket_object_content` data source

### ISSUE THREAD

**Issue by wafrelka** [hashicorp/terraform-provider-google#15717]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```shellsession
$ terraform -v
Terraform v1.5.6
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.80.0
+ provider registry.terraform.io/hashicorp/time v0.9.1
```

### Affected Resource(s)

* `google_storage_bucket_object_content`

### Terraform Configuration Files

```tf
terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "4.80.0"
    }
    time = {
      source  = "hashicorp/time"
      version = "0.9.1"
    }
  }
}

resource "time_sleep" "foo" {
  create_duration = "1s"
}

data "google_storage_bucket_object_content" "landsat" {
  bucket     = "gcp-public-data-landsat"
  name       = "LC08/01/001/003/LC08_L1GT_001003_20171007_20171023_01_T2/LC08_L1GT_001003_20171007_20171023_01_T2_ANG.txt"
  depends_on = [time_sleep.foo]
}

output "output" {
  value = replace(data.google_storage_bucket_object_content.landsat.content, "END", "end")
}
```

### Expected Behavior

`terraform apply` succeeds.

### Actual Behavior

`terraform apply` failed with the following error.

```
Error: Invalid function argument

  on main.tf line 25, in output "output":
  25:   value = replace(data.google_storage_bucket_object_content.landsat.content, "END", "end")
    ├────────────────
    │ while calling replace(str, substr, replace)
    │ data.google_storage_bucket_object_content.landsat.content is null

Invalid value for "str" parameter: argument must not be null.
```

Removing the `depends_on` attribute resolves the error.

### Steps to Reproduce

1. `terraform apply`

#### similar reports

- #14666

#### root causes

The [document](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/storage_bucket_object_content) says `content` is a *computed* attribute, but the implementation appears to disagree.
I suspect this is the root cause of this problem.

(The document also says it is "[Content-Language](https://tools.ietf.org/html/rfc7231#section-3.1.3.2) of the object content", but this also appears to be wrong.)

https://github.com/hashicorp/terraform-provider-google/blob/0b063b1c892decf057ef5a35954a1c436aab2cec/google/services/storage/data_source_storage_bucket_object_content.go#L23

https://github.com/hashicorp/terraform-provider-google/blob/0b063b1c892decf057ef5a35954a1c436aab2cec/google/tpgresource/datasource_helpers.go#L73-L75

https://github.com/hashicorp/terraform-provider-google/blob/0b063b1c892decf057ef5a35954a1c436aab2cec/google/tpgresource/datasource_helpers.go#L61-L67

**Comment by edwardmedia** [comment]
@wafrelka noticed below statement. Curious why you want to add `depends_on`?

```
Removing the depends_on attribute resolves the error.
```

**Comment by wafrelka** [comment]
I added `depends_on` to make it easy to reproduce.

Actually I encountered this problem while writing terraform config where the `google_storage_bucket_object_content` data source implicitly depends on a `google_storage_bucket` resource through the `bucket` attribute, like the example described in #14666. When the dependency `google_storage_bucket` needs to be updated, the `content` attribute becomes `null` in the plan phase.

debug log: https://gist.github.com/wafrelka/8a192df15855911533940f2a384bc730
